commit 09e5c6c2a3f8eac91d5353e6d4ff6aee7757ab08
Author: Steve Dickson <steved@redhat.com>
Date:   Mon Apr 24 11:25:39 2017 -0400

    systemd: Afters are also needed for the Wants=network-online.target
    
    Commit 9d4fc3fb added Wants=network-online.target which
    is not enough to ensure the network is completely up
    before the NFS server is started. After=network-online.target
    is also needed.
    
    Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1419351
    
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/systemd/nfs-mountd.service b/systemd/nfs-mountd.service
index fec0399..e8ece53 100644
--- a/systemd/nfs-mountd.service
+++ b/systemd/nfs-mountd.service
@@ -4,6 +4,7 @@ DefaultDependencies=no
 Requires=proc-fs-nfsd.mount
 Wants=network-online.target
 After=proc-fs-nfsd.mount
+After=network-online.target local-fs.target
 After=rpcbind.socket
 BindsTo=nfs-server.service
 
diff --git a/systemd/nfs-server.service b/systemd/nfs-server.service
index e6b8f58..136552b 100644
--- a/systemd/nfs-server.service
+++ b/systemd/nfs-server.service
@@ -7,7 +7,7 @@ Wants=rpcbind.socket network-online.target
 Wants=rpc-statd.service nfs-idmapd.service
 Wants=rpc-statd-notify.service
 
-After= local-fs.target
+After= network-online.target local-fs.target
 After= proc-fs-nfsd.mount rpcbind.socket nfs-mountd.service
 After= nfs-idmapd.service rpc-statd.service
 Before= rpc-statd-notify.service
diff --git a/systemd/rpc-statd-notify.service b/systemd/rpc-statd-notify.service
index f54d4c5..687fe31 100644
--- a/systemd/rpc-statd-notify.service
+++ b/systemd/rpc-statd-notify.service
@@ -2,7 +2,7 @@
 Description=Notify NFS peers of a restart
 DefaultDependencies=no
 Wants=network-online.target
-After=local-fs.target network.target nss-lookup.target
+After=local-fs.target network-online.target nss-lookup.target
 
 # if we run an nfs server, it needs to be running before we
 # tell clients that it has restarted.
diff --git a/systemd/rpc-statd.service b/systemd/rpc-statd.service
index 8cef022..f41ae20 100644
--- a/systemd/rpc-statd.service
+++ b/systemd/rpc-statd.service
@@ -4,7 +4,7 @@ DefaultDependencies=no
 Conflicts=umount.target
 Requires=nss-lookup.target rpcbind.socket
 Wants=network-online.target
-After=nss-lookup.target rpcbind.socket
+After=network-online.target nss-lookup.target rpcbind.socket
 
 PartOf=nfs-utils.service
 
